.github/workflows/test-pr.yml

```
@@ -11,9 +11,16 @@ jobs:
     - name: Install Poetry
       uses: snok/install-poetry@v1
     - run: poetry install
-    - run: poetry run coverage run -m pytest
+    - run: poetry run coverage run --branch -m pytest
+    - run: poetry run coverage json
```

I think adding names to these steps will make the action itself look better, like so:

```suggestion
    - name: Run unit tests and report coverage
      run: poetry run coverage run --branch -m pytest
    - name: Convert coverage results to JSON (for CodeCov support)
      run: poetry run coverage json
```